module/plugins/hooks/LinkFilter.py

```diff
@@ -22,18 +25,46 @@ def deactivate(self):
         self.manager.removeEvent('linksAdded', self.filter_links)
 
     def filter_links(self, links, pid):
-        filter_entries = self.config.get('filter').split(',')
+        filters = self.config.get('filter').replace(' ', '').split(',')
+        if self.config.get('whitelist'):
+            self.whitelist(links, filters)
+        else:
+            self.blacklist(links, filters)
 
-        for filter in filter_entries:
-            if filter == "":
-                break
+    def whitelist(self, links, filters):
+        linkcount = len(links)
+        links[:] = [link for link in links if
+            not self.isHosterLink(link) or
+            [True for filter in filters if link.find(filter) != -1]]
```

Comment:
Shouldn't that be all() or any()?

Reply:
I don't know if it should. Did some testing and it looks like
`[True for filter in filters if link.find(filter) != -1]`
is doing the same as
`any(True for filter in filters if link.find(filter) != -1)`

Reply:
This is not true:
```
>>> bool([True, False])
True
>>> bool([False, False])
True
>>> bool([True, True])
True
```
What it actually does is to check if the array is not empty which is not.

So:
`any(link.find(filter) != -1 for filter in filters)`
is  correct and also more readable..

Reply:
What I meant is that I tested both codes using 8 links from different sites (mostly hosters) and changed the plugin settings and filter all the time, then readded the links to queue and compared the outcome. And the filter function haad the same result with both code fragments (Also changed order of links while adding + in filter). However, I agree that the readibilty is better, so in any way it makes sense to change it.

Reply:
@segelkma Please test this version:
[LinkFilter.zip](https://github.com/pyload/pyload/files/1465255/LinkFilter.zip)